distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -29,21 +29,30 @@
 
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
 import java.util.Locale;
 
 /**
  * The main CLI for running Elasticsearch.
  */
 class ServerCli extends EnvironmentAwareCommand {
 
+    interface ServerProcessPostExitProcessor {
+        void process(int exitCode);
+
+        void waitForCompletion() throws InterruptedException;
```

Comment:
Q for reviewers: should we add the possibility to specify a timeout here?

Reply:
I don't know if we have timeouts at a layer above this - maybe in the container itself? If we do, I dont think we need another timeout here.

Reply:
I think so, and I agree with you: if we have a timeout at a higher level, no need to introduce another one